### ISSUE TITLE
IDE issues with project structure

### ISSUE THREAD

**Issue by milesfrain** [JordanMartinez/purescript-cookbook#230]
PureScript IDEs do not play nice with the cookbook directory structure.

The IDEs usually expect to find these in the root directory:
```
root
  output/
  packages.dhall
  spago.dhall
```

We still have a common `output/` and `packages.dhall` in root, but use a separate `spago.dhall` for each recipe.
```
root
  output/
  packages.dhall
  recipes
    RecipeFoo
      spago.dhall
    RecipeBar
      spago.dhall
```
So if you open your editor at the root, you'll see errors about a missing `spago.dhall`.

One workaround is to change your development root (where you open your IDE from) to the directory of the recipe you are editing:
```
cd recipes/RecipeFoo
spago build
code .
```

This workaround has a few issues:
- `output/` is duplicated in the recipe directory.
- Automatic page refresh (from `make RecipeFoo-web`) will not pick-up this local `output/`, since it relies on `output/` at root.
  - You can get page refresh to work by running `make RecipeFoo-build-watch` in another terminal, which will also build changes into `output/` at root. This is not ideal, because now we're duplicating effort by building into two `output/` locations.

**Comment by milesfrain** [comment]
A follow-up task is to document our recommended workaround in Contributing.md. Will do that after #228.
